### PR TITLE
update PluginManager

### PR DESCRIPTION
**Pull request by shitake2333**
Provide a new way to create plugin command like `commander.js`.
@niokasgami 
@liply 
#40

### CODE REVIEW

File: plugin/Example_Plugin.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * Plugin command: 
+ * 
+ * example init helloworld 12 12 640 480
+ * 
+ * output:
+ *  Plugin: Exmaple, Command: init, Params: {title: helloworld, x: 12, y: 12, width: 640, height: 480}
+ */
+program.command('init', 'title:string, x:number, y:number, width:number, height:number', function(params) {
```
Comment (ghost) [inline plugin/Example_Plugin.js:28]: Why would you supply the parameters as a string here, when you just split them up at "," again later?
Wouldn't it make more sense to pass a dictionary right away, e.g. {"title":"string", "x":"number", "y":"number",...} ?

File: js/rpg_managers/PluginManager.js [hunk 2]
```diff
@@ -51,3 +52,91 @@ PluginManager.loadScript = function(name) {
...
+    this.initialize.apply(this, arguments);
+}
+
+PluginProgram.prototype.initialize = function(name) {
+    this._name = name;
+    this._commands = {};
+}
+
+PluginProgram.createParams = function(paramsStyle, args) {
+    if(paramsStyle.length != args.length) throw new Error('Wrong number of arguments (given ' + args.length + ', expected ' + paramsStyle.length + ').')
```
Comment (ghost) [inline js/rpg_managers/PluginManager.js:86]: This is just a minor readability nitpick, but this line seems needlessly long for no good reason. I would recommend writing this as four lines:

```javascript
if (paramsStyle.length != args.length) {
    throw new Error('Wrong number of arguments (given ' + args.length +
        ', expected ' + paramsStyle.length + ').');
}
```

File: js/rpg_managers/PluginManager.js [hunk 3]
```diff
@@ -51,3 +52,91 @@ PluginManager.loadScript = function(name) {
...
+    var params = {};
+    args.forEach(function(arg, index) {
+        var value;
+        switch (paramsStyle[index].type) {
+            case 'number':
+                value = Number(arg)
```
Comment (ghost) [inline js/rpg_managers/PluginManager.js:92]: You might want to add a `;` here, just for consistency

File: plugin/Example_Plugin.js (hunk 1 above)
Comment (shitake2333) [reply plugin/Example_Plugin.js:28]: I think the two versions can be. 
By way, if we use Object here, so write `{title: String, x: Number, y:Number, width: Number, height: Number}  better.

**Comment by shitake2333** [comment]
@JonasT 
This PR is just a few of my ideas for the discussion of the plugin manager. So the code is relatively rough. Later I will update a more complete version.

**Pull request closed without merging**